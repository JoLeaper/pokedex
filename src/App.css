.search {
    display: flex;
}

.searchbox {
    display: flex;
    background-color: gray;
    width: 20%;
    height: 750px;
    flex-direction: column
}

.resultsBox {
    width: 80%;
    background-color: blue;
    display: grid;
    grid-template-columns: 1fr 1fr
}