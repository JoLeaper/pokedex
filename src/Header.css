.header{
    background-color: red;
    color: white;
    height: 100px;
    padding-top: 1px;
    margin: 0;
    position: relative;
}

.title {
    position: fixed;
    left: 2%;

}

.nav {
    position: fixed;
    right: 2%;
}