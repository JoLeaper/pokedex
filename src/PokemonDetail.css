.detailPokemon {
    border: 2px solid black;
    background-color: lightgray;
    margin: 0px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: center;
    width: auto;
    height: auto;
}
