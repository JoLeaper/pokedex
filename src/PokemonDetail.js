import React, { Component } from 'react'
import request from 'superagent';

export default class PokemonDetail extends Component {
    state = {
        pokemon: null
    };
    componentDidMount = async () => {
        const pokemonId = this.props.match.params.id
        const fetchedPokemon = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex/${pokemonId}`);
        this.setState({
            pokemon: fetchedPokemon.body
        }
        )
        console.log(this.state.pokemon);
    }
    render() {
        return this.state.pokemon ?
            <div>
                <h2>{this.state.pokemon.pokemon}</h2>
                <img src={this.state.pokemon.url_image} alt={this.state.pokemon.pokemon} />
                <p>Type: {this.state.pokemon.type_1}  {(this.state.pokemon.type_2 === 'NA')
                    ? null
                    : '/ ' + this.state.pokemon.type_2}</p>
                <p>Ability: {this.state.pokemon.ability_1}</p>
                <p>Hidden Ability: {this.state.pokemon.ability_2}</p>
                <p>Base HP: {this.state.pokemon.hp}</p>
                <p>Base Attack: {this.state.pokemon.attack}</p>
                <p>Base Defense: {this.state.pokemon.defense}</p>
                <p>Base Special Attack: {this.state.pokemon.special_attack}</p>
                <p>Base Special Defense: {this.state.pokemon.special_defense}</p>
                <p>Base Speed: {this.state.pokemon.speed}</p>
            </div>
            :
            <img src='https://i.giphy.com/media/jQbinPsqqfg8GFxbQw/giphy.webp' alt='loading'></img>
    }
}
