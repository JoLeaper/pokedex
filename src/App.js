import React, { Component } from 'react'
import Header from './Header.js'
import SearchResults from './SearchResults.js'
import request from 'superagent';
import './App.css'

export default class App extends Component {
  state = {
    displayOrder: 'asc',
    pokemonName: '',
    pokemonType: ['none', 'normal', 'fire', 'fighting', 'water', 'flying', 'grass', 'poison', 'electric', 'ground', 'psychic', 'rock', 'ice', 'bug', 'dragon', 'ghost', 'dark', 'steel', 'fairy'],
    desiredPokemonType: '',
    pokemonAttack: 0,
    pokemon: []
  }

  handleOrderChange = (event) => {
    this.setState({
      displayOrder: event.target.value
    })
  }

  handleTypeChange = (event) => {
    this.setState({
      desiredPokemonType: event.target.value
      
    })
    console.log(this.state.pokemonType)
  }

  handleNameChange = (event) => {
    this.setState({
      pokemonName: event.target.value
    })
  }

  handleAttackChange = (event) => {
    this.setState({
      pokemonAttack: event.target.value
    })
  }

  handleClick = async () => {
    const searchedPokemon = this.state.pokemonName;
    const minAttack = this.state.pokemonAttack;
    let wantedType;
    if (this.state.desiredPokemonType !== 'none') {
      wantedType='&type=' + this.state.desiredPokemonType
    }

    const fetchedPokemon = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex?pokemon=${searchedPokemon}${wantedType}&attack=${minAttack}&sort=id&direction=${this.state.displayOrder}`)
    this.setState({ pokemon: fetchedPokemon.body.results })
  }

  render() {
    return (
      <div className='container'>
        <Header />
        <div className='search'>
          <div className='searchbox'>
            Name:
            <input value={this.state.PokemonName} onChange={this.handleNameChange}>
            </input>
              Display Order (By Pokedex Number):
              <select value={this.state.displayOrder} onChange={this.handleOrderChange}>
                <option value='asc'>Early Pokemon</option>
                <option value='desc'>Later Pokemon</option>
                
              </select>
              Type:
              <select onChange={this.handleTypeChange}>
                {
                this.state.pokemonType.map(type => 
                <option value={type}>{type}</option>
                 )
                 }
              </select>

              Minimum Attack Of:<input value={this.state.pokemonAttack} onChange={this.handleAttackChange}>
              </input>
              <button onClick={this.handleClick}>Search</button>
        </div>
        
          <SearchResults pokemonList={this.state.pokemon}/>


        </div>
      </div>
    )
  }
}
