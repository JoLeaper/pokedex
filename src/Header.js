import React, { Component } from 'react'
import './Header.css'

export default class Header extends Component {
    render() {
        return (
            <div className='header'>
                <h1 className='title'>Welcome to the Pokedex!</h1>
                {/* <h1 className='nav'>Nav</h1> */}
            </div>
        )
    }
}
