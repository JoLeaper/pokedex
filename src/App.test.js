import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import Header from './Header';
import PokemonDetail from './PokemonDetail'

test('renders learn react link', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

test('renders learn react link', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});

test('renders learn react link', () => {
  const wrapper = shallow(<PokemonDetail />);
  expect(wrapper).toMatchSnapshot();
});


