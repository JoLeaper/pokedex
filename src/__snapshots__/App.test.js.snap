// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="search"
>
  <SearchBar
    displayOrder="asc"
    handleAttackChange={[Function]}
    handleClick={[Function]}
    handleNameChange={[Function]}
    handleOrderChange={[Function]}
    handlePageClick={[Function]}
    handleTypeChange={[Function]}
    pokemonAttack={0}
    pokemonName={null}
    pokemonType={
      Array [
        "",
        "normal",
        "fire",
        "fighting",
        "water",
        "flying",
        "grass",
        "poison",
        "electric",
        "ground",
        "psychic",
        "rock",
        "ice",
        "bug",
        "dragon",
        "ghost",
        "dark",
        "steel",
        "fairy",
      ]
    }
  />
  <SearchResults
    pokemonList={Array []}
  />
</div>
`;
